spec/models/user_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { should validate_presence_of :uid }
+  it { should validate_presence_of :login }
+  it { should validate_presence_of :email }
+  it { should validate_uniqueness_of(:uid) }
+  it { should validate_uniqueness_of(:login) }
+
+  describe 'from_omniauth' do
+    let(:user) { create(:user) }
+    let(:auth) do  OmniAuth::AuthHash.new(
+      provider: user.provider,
+      uid:      user.uid,
+      info:     { email: user.email, name: user.name },
+      extra:    { raw_info: { login: user.login }}
+    )
+    end
+
+    context 'the user has not authorization' do
```

Comment:
You have your test with `create` — so here you're logging in as a user already in the DB.
I would suggest you to test a second branch, and use `build(:user)` instead of creating it — so you can ensure that user is actually created in the DB.